### PR TITLE
docs: update Go API page

### PR DESCRIPTION
**Pull request by jmcook1186**
updates content on the Go API overview page, including replacing the screenshot used as a clickable link to Youtube to something a little slicker.

### CODE REVIEW

File: docs/_dapp/native.md [hunk 1]
```diff
@@ -3,64 +3,89 @@ title: Go API
...
 
 ## Go packages
 
-The `go-ethereum` library is distributed as a collection of standard Go packages straight
-from our GitHub repository. The packages can be used directly via the official Go toolkit,
-without needing any third party tools. External dependencies are vendored locally into
-`vendor`, ensuring both self-containment as well as code stability. If you reuse
-`go-ethereum` in your own project, please follow these best practices and vendor it
-yourself too to avoid any accidental API breakages!
+The Geth library is distributed as a collection of standard Go packages straight from Geth's GitHub
```
Comment (holiman) [inline docs/_dapp/native.md:60]:
```suggestion
The `go-ethereum` library is distributed as a collection of standard Go packages straight from Geth's GitHub
```

File: docs/_dapp/native.md [hunk 2]
```diff
@@ -3,64 +3,89 @@ title: Go API
...
+repository. The packages can be used directly via the official Go toolkit, without needing any 
+third party tools. External dependencies are vendored locally into `vendor`, ensuring both 
+self-containment and code stability. When Geth is used in downstream projects these best
+practices (packing dependencies into a local `vendor`) should be followed there too to 
+avoid any accidental API breakages.
```
Comment (holiman) [inline docs/_dapp/native.md:65]:
```suggestion
third party tools.
```
The 'vendor' was something that predated go modules, which now is fully supported and makes a lot of the problems around external dependencies go away. Or, at least become a lot more standardized.

File: docs/_dapp/native.md [hunk 3]
```diff
@@ -3,64 +3,89 @@ title: Go API
...
+The libraries are updated synchronously with the Geth Github repository. 
+The Go libraries can be viewed in full at [Go Packages][go-pkg-link]
+
+The benefits of using the Go libraries include easy client-side [account management](/docs/dapp/native-accounts)
+that allows users to keep their private keys safely encrypted locally and make their own
+security decisions. The Go libraries allow dapp developers to spend more of their time
+building their dapps rather than implementing complex protocols such as encoding and
+decoding function calls to smart contracts - this can all be abstracted away using
+Geth's Go libraries.
+
```
Comment (holiman) [inline docs/_dapp/native.md:50]: This looks to me like a sales pitch. I don't see why we need to pitch people to use our libraries for encoding/decoding.
 
I'd prefer less text, and would rather remove blurbs like these which doesn't provide information.

File: docs/_dapp/native.md [hunk 4]
```diff
@@ -3,64 +3,89 @@ title: Go API
...
+transact between accounts and interact with smart contracts using command line tools. 
+Geth was one of the original clients to provide this type of gateway to the Ethereum network.
 
-The first incarnation towards this dream of web3 was a command line client providing an
-RPC interface into the peer-to-peer protocols. The client was soon enough extended with a
-web-browser-like graphical user interface, permitting developers to write DApps based on
-the tried and proven HTML/CSS/JS technologies.
+Before long, web-browser-like graphical interfaces (e.g. Mist) were created to extend clients.
+However, as dapps became more complex their requirements exceeded what a simple browser environment
+could support. To support the next generation of dapps, built using time-tested HTML/CSS/JS
```
Comment (holiman) [inline docs/_dapp/native.md:15]: Heh, this isn't your sentence, but it's IMO a pretty bad sentence even from the beginning. "what a simple browser environment could support" -- a modern browser environment can support some pretty darn complex things.
 
I'm not sure how to better phrase it , but programmatic access to a larger/deeper API is at the core, I think.

Comment (jmcook1186) [reply docs/_dapp/native.md]: ok I agree it is clunky. Will try to reword.

**Comment by jmcook1186** [comment]
ok I agree that would be nice to have - I will look into it and ping you with some questions later!

**Review by s1na — APPROVED** [review]
Made some modifications and added a filter logs example. LGTM.

At some point we can also add examples for the subscription methods.

**Pull request merged**